pkg/core/resources/store/pagination_store.go

```diff
@@ -46,7 +46,7 @@ func (p *paginationStore) List(ctx context.Context, list model.ResourceList, opt
 	opts := NewListOptions(optionsFunc...)
 
 	// Performance optimization
```

Comment:
This comment is unclear. Please explain this is an early return if no filter or pagination is set or something like this